File: acapella_manager/lib/services/auth_service.dart
import 'package:dio/dio.dart';
import '../models/user.dart';
import 'api_client.dart';

class AuthService {
  final Dio _dio = ApiClient().dio;

  Future<Map<String, dynamic>> login(String email, String password) async {
    final response = await _dio.post('/auth/login', data: {
      'email': email,
      'password': password,
    });

    if (response.statusCode == 200) {
      final data = response.data['data'];
      ApiClient().setToken(data['token']);
      return {
        'user': User.fromJson(data['user']),
        'token': data['token'],
      };
    }
    throw Exception('Login failed.');
  }

  Future<void> logout() async {
    await _dio.post('/auth/logout');
    ApiClient().clearToken();
  }

  Future<User> getMe() async {
    final response = await _dio.get('/auth/me');
    if (response.statusCode == 200) {
      return User.fromJson(response.data['data']);
    }
    throw Exception('Failed to load profile.');
  }
}

File: acapella_api_2/resources/js/Services/authService.js
import apiService, { STORAGE_KEYS, extractData } from './apiService';
import { db } from '../Utils/indexedDB';

const persistSession = async (payload) => {
    const data = payload?.data ?? payload;
    const token = data?.token;
    const user = data?.user_info || data?.user || data;

    if (token) {
        await db.set('auth', STORAGE_KEYS.AUTH_TOKEN, token);
    }

    if (user) {
        await db.set('auth', STORAGE_KEYS.USER_DATA, user);
    }

    return { token, user };
};

const authService = {
    async register(payload) {
        const response = await apiService.post('/auth/register', payload);
        return persistSession(extractData(response));
    },

    async login(credentials) {
        const response = await apiService.post('/auth/login', credentials);
        return persistSession(extractData(response));
    },

    async logout() {
        try {
            await apiService.post('/auth/logout');
        } finally {
            await db.delete('auth', STORAGE_KEYS.AUTH_TOKEN);
            await db.delete('auth', STORAGE_KEYS.USER_DATA);
        }
    },

    async me() {
        const response = await apiService.get('/auth/me');
        const user = extractData(response);
        await db.set('auth', STORAGE_KEYS.USER_DATA, user);
        return user;
    },

    async forgotPassword(email) {
        const response = await apiService.post('/auth/forgot-password', { email });
        return extractData(response);
    },

    async resetPassword(payload) {
        const response = await apiService.post('/auth/reset-password', payload);
        return extractData(response);
    },

    async getToken() {
        await db.init();
        return await db.get('auth', STORAGE_KEYS.AUTH_TOKEN);
    },

    async getCurrentUser() {
        await db.init();
        const user = await db.get('auth', STORAGE_KEYS.USER_DATA);
        return user || null;
    },

    async clearSession() {
        await db.delete('auth', STORAGE_KEYS.AUTH_TOKEN);
        await db.delete('auth', STORAGE_KEYS.USER_DATA);
    },
};

export default authService;
